transform-plugins/src/test/java/co/cask/hydrator/plugin/XMLParserTest.java

```diff
@@ -0,0 +1,315 @@
+
+  @Test
+  public void testInvalidConfig() throws Exception {
+    XMLParser.Config config = new XMLParser.Config("body", "UTF-8", "category://book/@category,title://book/title," +
+      "year:/bookstore/book[price>35.00]/year,price:/bookstore/book[price>35.00]/price,subcategory://book/subcategory",
+                                                   "category:,title:string,price:double,year:int,subcategory:string",
+                                                   "Exit on error");
+
+    MockPipelineConfigurer configurer = new MockPipelineConfigurer(INPUT);
+    try {
+      new XMLParser(config).configurePipeline(configurer);
+    } catch (IllegalArgumentException e) {
+      Assert.assertEquals("Type cannot be null. Please specify type for category", e.getMessage());
+    }
+  }
+
+  @Test
+  public void testXMLParserWithSimpleXPath() throws Exception {
+    String inputTable = "input-simple-xpath";
+    ETLStage source = new ETLStage("source", MockSource.getPlugin(inputTable));
+
+    Map<String, String> sourceProperties = new ImmutableMap.Builder<String, String>()
+      .put("input", "body")
+      .put("encoding", "UTF-8")
+      .put("xPathMappings", "title:/book/title,author:/book/author,year:/book/year")
+      .put("fieldTypeMapping", "title:string,author:string,year:string")
+      .put("processOnError", "Write to error dataset")
+      .build();
+
+    ETLStage transform = new ETLStage("transform",
+                                      new ETLPlugin("XMLParser", Transform.PLUGIN_TYPE, sourceProperties, null));
+    String sinkTable = "output-simple-xpath";
+
+    ETLStage sink = new ETLStage("sink", MockSink.getPlugin(sinkTable));
+    ETLBatchConfig etlConfig = ETLBatchConfig.builder("* * * * *")
+      .addStage(source)
+      .addStage(transform)
+      .addStage(sink)
+      .addConnection(source.getName(), transform.getName())
+      .addConnection(transform.getName(), sink.getName())
+      .build();
+
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(ETLBATCH_ARTIFACT, etlConfig);
+    ApplicationId appId = NamespaceId.DEFAULT.app("XMLReaderTest");
+    ApplicationManager appManager = deployApplication(appId.toId(), appRequest);
+
+    DataSetManager<Table> inputManager = getDataset(inputTable);
+    List<StructuredRecord> input = ImmutableList.of(
+      StructuredRecord.builder(INPUT)
+        .set("offset", 1)
+        .set("body", "<book category=\"COOKING\"><title lang=\"en\">Everyday Italian</title>" +
+          "<author>Giada De Laurentiis</author><year>2005</year><price>30.00</price></book>").build()
+    );
+    MockSource.writeInput(inputManager, input);
+    MapReduceManager mrManager = appManager.getMapReduceManager(ETLMapReduce.NAME);
+    mrManager.start();
+    mrManager.waitForFinish(5, TimeUnit.MINUTES);
+
+    DataSetManager<Table> outputManager = getDataset(sinkTable);
+    List<StructuredRecord> outputRecords = MockSink.readOutput(outputManager);
```

Comment:
We are asserting the outputRecords.size() in line #116.
Please let me know if this is okay or we need to check any extra parameters.